test/Altinn.App.Api.Tests/Middleware/ProcessLockMiddlewareTests.cs

```diff
@@ -0,0 +1,388 @@
+                    .WithBodyAsJson(new ProcessLockResponse { LockId = lockId })
+            );
+
+        fixture
+            .Server.Given(fixture.GetReleaseLockRequestBuilder(lockId))
+            .RespondWith(Response.Create().WithStatusCode(HttpStatusCode.InternalServerError));
+
+        var response = await fixture
+            .GetTestClient()
+            .GetAsync($"/instances/{fixture.InstanceOwnerPartyId}/{fixture.InstanceGuid}/test");
+
+        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+        var content = await response.Content.ReadAsStringAsync();
+        Assert.Equal("Success", content);
+
+        var releaseRequests = fixture.Server.FindLogEntries(fixture.GetReleaseLockRequestBuilder(lockId));
+        Assert.Single(releaseRequests);
+    }
+
+    [Theory]
+    [InlineData(HttpStatusCode.Conflict)]
+    [InlineData(HttpStatusCode.NotFound)]
+    [InlineData(HttpStatusCode.InternalServerError)]
+    public async Task StorageApiError_ReturnsCorrectStatusCode(HttpStatusCode storageStatusCode)
+    {
+        using var fixture = Fixture.Create();
+
+        fixture
+            .Server.Given(fixture.GetAcquireLockRequestBuilder())
+            .RespondWith(Response.Create().WithStatusCode(storageStatusCode));
+
+        var response = await fixture
+            .GetTestClient()
+            .GetAsync($"/instances/{fixture.InstanceOwnerPartyId}/{fixture.InstanceGuid}/test");
+
+        Assert.Equal(storageStatusCode, response.StatusCode);
+
+        var problemDetails = await response.Content.ReadFromJsonAsync<ProblemDetails>();
+
+        Assert.NotNull(problemDetails);
+        Assert.Equal("Failed to acquire lock.", problemDetails.Title);
+        Assert.Equal((int)storageStatusCode, problemDetails.Status);
+
+        Assert.Single(fixture.Server.LogEntries);
+    }
+
+    [Fact]
+    public async Task NullResponseBody_ReturnsProblemDetails()
+    {
+        using var fixture = Fixture.Create();
+
+        fixture
+            .Server.Given(fixture.GetAcquireLockRequestBuilder())
+            .RespondWith(
+                Response
+                    .Create()
+                    .WithStatusCode(HttpStatusCode.OK)
+                    .WithHeader("Content-Type", "application/json")
+                    .WithBody("null")
+            );
+
+        var response = await fixture
+            .GetTestClient()
+            .GetAsync($"/instances/{fixture.InstanceOwnerPartyId}/{fixture.InstanceGuid}/test");
+
+        Assert.Equal(HttpStatusCode.InternalServerError, response.StatusCode);
+
+        var problemDetails = await response.Content.ReadFromJsonAsync<ProblemDetails>();
+
+        Assert.NotNull(problemDetails);
+        Assert.Equal("Failed to acquire lock.", problemDetails.Title);
+        Assert.Equal("The response from the lock acquisition endpoint was not expected.", problemDetails.Detail);
+        Assert.Equal((int)HttpStatusCode.InternalServerError, problemDetails.Status);
```

Comment:
These kinds of tests are good candidates for snapshot testing I think, creates less churn if we decide to change wording etc (snapshots are just automatically updated, don't have to refactor assertions)